Sources/HttpPipeline/Response.swift

```diff
@@ -1,7 +1,146 @@
 import Foundation
+import MediaType
+import Prelude
 
 public struct Response {
   public private(set) var status: Status
-  public private(set) var headers: [ResponseHeader]
+  public private(set) var headers: [Header]
   public private(set) var body: Data
+
+  public struct Header {
```

Comment:
Nesting this now (instead of hiding away in another file).

`ResponseHeader` used to be an `enum`. Now it's a `struct`. This means that we eagerly convert values to strings. (Is this potentially extra work for a failing request? Is this worth making lazy?) This also means we can't pattern-match anymore, which is fine, since `case let .contentType(t), let .other("Content-Type", t)` doesn't really work.